### DISCUSSION TITLE
E2 mix functions implemented backwards

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2893] [Suggestions]
https://github.com/wiremod/wire/blob/ccfd65ee6bf597e8d2349da78c2e13362284f6ad/lua/entities/gmod_wire_expression2/core/vector.lua#L524-L530

`mix(vvn)` is implemented so that `mix(vec1, vec2, 0) == vec2`. The same applies to all other mix functions (vector2, vector4). This is confusing compared to typical definitions of similar functions.

The expected behavior is `mix(vec1, vec2, 0) == vec1`.

Not sure how no-one has complained about this before.

**Updates:**
This thread is now a bit outdated for mentioning vectors specifically, but this still holds true for other types, so those still need to be addressed.
#2895: Deprecated `mix(vector, vector, number)`

**Comment by vurvdev** [comment]
With https://github.com/wiremod/wire/discussions/2721, could probably fix this ordering issue with the `color` type and deprecate this function. And then maybe add `lerp(vvn)` too.

**Comment by Fasteroid** [comment]
Agreed, it's definitely backwards, and I've known for ages.  I just never complained because fixing something like this requires either deprecating functions or breaking backwards-compat (lol)

**Comment by Hazeofdream** [comment]
when e2 literally breaks chips by fixing its self lmao

does this mean e2 functions by being broken

**Comment by vurvdev** [comment]
Converted to a discussion since this isn't a bug just bad design
